File: packages/authentication/ios/Plugin/Classes/GetIdTokenResult.swift
import Foundation
import FirebaseAuth
import Capacitor

@objc class GetIdTokenResult: NSObject {
    let token: String

    init(token: String) {
        self.token = token
    }

    func toJSObject() -> JSObject {
        var result = JSObject()
        result["token"] = self.token
        return result
    }
}

File: packages/authentication/android/src/main/java/app/capgo/capacitor/firebase/authentication/GetIdTokenResult.java
package app.capgo.capacitor.firebase.authentication.classes;

import androidx.annotation.Nullable;
import app.capgo.capacitor.firebase.authentication.interfaces.Result;
import com.getcapacitor.JSObject;

public class GetIdTokenResult implements Result {

    @Nullable
    private String token;

    public GetIdTokenResult(@Nullable String token) {
        this.token = token;
    }

    @Nullable
    public String getToken() {
        return token;
    }

    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("token", token);
        return result;
    }
}
